### PR TITLE
fix(hc): Make IntegrationService utility methods non-RPC

### PR DESCRIPTION
**Pull request by RyanSkonnord**

### CODE REVIEW

**Comment by RyanSkonnord** [comment]
@leeandher Do I have it right that these are just utility methods, and we expect both of them to return the same values on both the control and region silos? My understanding is that they're both just talking to the `IntegrationManager` and not the DB.

Or, were they actually intended to go over the wire? There are some cases for using `@rpc_method` without `@abstractmethod` but I don't think one applies here.

File: src/sentry/services/hybrid_cloud/integration/__init__.py [hunk 1]
```diff
@@ -299,7 +299,6 @@ def update_organization_integration(
 
     # The following methods replace instance methods of the ORM objects!
 
-    @rpc_method
     def get_installation(
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/integration/__init__.py:302]: This one has to go over the wire, as Installations are only in Control Silo. Perhaps we can modify `provider.get_installation()` to take an id instead?

Comment (leeandher) [reply src/sentry/services/hybrid_cloud/integration/__init__.py:302]: When I put together the RPC methods for integrations, I believe the installations call the RPC, meaning calling get_installtion should be silo agnostic.

That means that `provider.get_installation().org_integration` [will make an RPC call](https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/base.py#L271-L278)

**Review by leeandher — APPROVED** [review]
I believe both these methods get values from or instantiate in-memory python objects, so I think they're safe. It might be worth noting that these objects may themselves invoke the RPC (e.g. `install.org_integration`) but removing these decorators should be safe.

**Pull request merged**